### PR TITLE
LPS-109025 Update react-dnd to v10.0.1

### PR DESCRIPTION
**Pull request by julien**

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-react-web/package.json [hunk 1]
```diff
@@ -5,8 +5,10 @@
 		"formik": "1.4.3",
 		"prop-types": "15.7.2",
 		"react": "16.12.0",
-		"react-dnd": "7.7.0",
-		"react-dnd-html5-backend": "7.7.0",
+		"react-dnd": "10.0.1",
+		"react-dnd-html5-backend": "10.0.1",
+		"react-dnd-test-backend": "10.0.1",
+		"react-dnd-test-utils": "10.0.1",
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-react-web/package.json]: These should be in `devDependencies`, looks like the `--dev` switch was missing, and the old versions are still present on lines 15/16 below (and you may want to regen the yarn.lock, not sure).

Comment (julien) [reply modules/apps/frontend-js/frontend-js-react-web/package.json]: :+1:

**Comment by andresfulla** [comment]
looks good to me. Thanks!

**Comment by john-co** [comment]
ci:test

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Hi @julien and @wincent, just wanted to raise a concern that this could be risky during stabilization phase of GA2 release. If not urgent or a requirement for GA2, perhaps we can wait until after release to merge?

**Comment by john-co** [comment]
1. Manual Testing Add Content Page on IE11
Navigate to Site Builder > Site Pages > Add Blank Page, Go to Edit Content Page view, Assert page loads indefinitely and possible error in browser dev tool is thrown:
image

**Comment by wincent** [comment]
That second one might be due to the changes in `modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/App.es.js`.

On another note, @julien, would you be able to `git commit --amend` this to include information about some of the aspects of this change that might be puzzling if we come looking at it days/weeks/months from now via `git blame` wanting to know what's up with the `resolutions` field etc? Something like:

```
Changes to `.npmbundlerrc` are needed because react-dnd updated
its "main" field to point at `dist` rather than `lib` about seven months
ago, in:

https://github.com/react-dnd/react-dnd/commit/86a13e9e822ad70370c9886cd00d529bd2ed0b67

We need to use "resolutions" to force the use of 10.0.1 instead of the
latest, 10.0.2, in order to avoid this problematic commit which forces
the use of a fork of the `asap` module:

https://github.com/react-dnd/react-dnd/commit/ada65550b758a5de44b810a7b68753e7eaee24ac

Which leads to an untransformed `require('domain')` call winding up
in the build output and breaking everything:

https://github.com/react-dnd/asap/blob/b6bebeb73428474aefa8250ac79cf0f81fd146d9/src/node/raw.ts#L87
```

Otherwise, I pity the poor fool who has to update this next time around.

**Comment by julien** [comment]
@wincent Thanks for the suggestion, I updated the commit message - and guessing what was wrong with `modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/App.es.js.`

@john-co Firefox errors have been fixed on Firefox. However for IE 11 I have a doubt

I asked the echo team if they had tested the new Page Editor in IE 11, and they reported that they couldn't because it also has errors - see [LPS-109074](https://issues.liferay.com/browse/LPS-109074). I went ahead and removed all references to `DOMRect` in my local Page Editor module, re-deployed and was still unable to load it.

I then checked out the master branch made sure I was up to date and did another ant all, and couldn't get the Page Editor to load in IE 11. 

IE11_02_after_fix


In anycase, thanks for your help, it might be worth testing again if you have the time do so.

I also spoke to @wincent and appart from the IE 11 error in Page Editor we think the rest of the changes are "safe" related to GA2.

**Comment by julien** [comment]
ci:test:bundle

**Comment by john-co** [comment]
ci:test

**Comment by john-co** [comment]
ci:test

**Comment by john-co** [comment]
- PASSED Manual Testing summarized in [LRQA-55974](https://issues.liferay.com/browse/LRQA-55974?focusedCommentId=2087635&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2087635) 
- Rerunning ci test.

**Comment by john-co** [comment]
Reviewed. Test failures are unrelated. :heavy_check_mark: 

**Comparison Details:**
- [Unique test failures](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=1014735896&testrayRunIdA=1015798345&statusB=2): Not related, flaky or ci server crashed
- [In common test failures](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=1014735896&testrayRunIdA=1015798345&statusB=3): Run A failures do not look related

**Comment by julien** [comment]
Sent to https://github.com/brianchandotcom/liferay-portal/pull/85147
But closed as requested by @carloslancha who will send his miller columns related changes on top

**Pull request closed without merging**